[container] v0.6.4/v0.7.0: nil pointer dereference in cri.go with containerd v2.3.0

**Describe the bug**

The container plugin panics with a nil pointer dereference in `cri.go:243` when running with containerd v2.3.0.

In `plugins/container/go-worker/pkg/container/cri.go:243`, the code accesses `cniInfo.RuntimeSpec.Annotations` without checking if `RuntimeSpec` is nil:

```go
} else if val, ok := cniInfo.RuntimeSpec.Annotations["io.kubernetes.cri-o.CNIResult"]; ok {
    cniJson = val
}
```

Containerd v2.3.0 (CRI API v0.36) does not populate the `runtimeSpec` field in sandbox info JSON — this field is cri-o specific. When `CNIResult` is also nil/empty, the code falls through to this branch and panics.

**How to reproduce it**

1. Run Falco with the container plugin v0.6.4 (or v0.7.0) on a node using containerd v2.3.0
2. Start Falco — the plugin crashes immediately during container listing

**Expected behaviour**

The plugin should gracefully handle a nil `RuntimeSpec` and skip the cri-o annotation lookup when running on containerd.

**Screenshots**

N/A

**Environment**

- Falco version: 0.43.1
- Cloud provider or hardware configuration: Bare metal
- OS: SUSE Linux Enterprise 15
- Installation method: Kubernetes DaemonSet

**Additional context**

Stack trace:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7fbe37302aef]

goroutine 17 [running, locked to thread]:
github.com/falcosecurity/plugins/plugins/container/go-worker/pkg/container.(*criEngine).ctrToInfo(...)
    cri.go:243 +0x76f
github.com/falcosecurity/plugins/plugins/container/go-worker/pkg/container.(*criEngine).List(...)
    cri.go:443 +0x391
main.StartWorker(...)
    worker_api.go:77 +0x4e2
```

Suggested fix — add a nil guard before accessing `RuntimeSpec.Annotations`:

```diff
--- a/plugins/container/go-worker/pkg/container/cri.go
+++ b/plugins/container/go-worker/pkg/container/cri.go
@@ -240,8 +240,10 @@
 			if err != nil {
 				cniJson = string(bytes)
 			}
-		} else if val, ok := cniInfo.RuntimeSpec.Annotations["io.kubernetes.cri-o.CNIResult"]; ok {
-			cniJson = val
+		} else if cniInfo.RuntimeSpec != nil {
+			if val, ok := cniInfo.RuntimeSpec.Annotations["io.kubernetes.cri-o.CNIResult"]; ok {
+				cniJson = val
+			}
 		}
```

This is a 100% crash rate — all Falco DaemonSet pods go into CrashLoopBackOff. The issue did not occur with containerd v2.2.3.

- Containerd: v2.3.0 (crashes) vs v2.2.3 (works)
- Container plugin: v0.6.4 (also present in v0.7.0 source)
- Kubernetes: v1.36.1

--- a/plugins/container/go-worker/pkg/container/cri.go
+++ b/plugins/container/go-worker/pkg/container/cri.go
@@ -240,8 +240,10 @@ func (c *criEngine) ctrToInfo(ctx context.Context, ctr *v1.ContainerStatus, podS
 				if err != nil {
 					cniJson = string(bytes)
 				}
-			} else if val, ok := cniInfo.RuntimeSpec.Annotations["io.kubernetes.cri-o.CNIResult"]; ok {
-				cniJson = val
+			} else if cniInfo.RuntimeSpec != nil {
+				if val, ok := cniInfo.RuntimeSpec.Annotations["io.kubernetes.cri-o.CNIResult"]; ok {
+					cniJson = val
+				}
 			}
 
 			if len(cniJson) > maxCNILen {

